go.mod

```diff
@@ -18,37 +18,37 @@ require (
 	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45
 	golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e
 	golang.org/x/sys v0.0.0-20191113165036-4c7a9d0fe056
-	google.golang.org/grpc v1.25.1
-	k8s.io/api v0.17.5
-	k8s.io/apimachinery v0.17.5
-	k8s.io/client-go v0.17.5
+	google.golang.org/grpc v1.26.0
+	k8s.io/api v0.18.3
+	k8s.io/apimachinery v0.18.3
+	k8s.io/client-go v0.18.3
 	k8s.io/kubernetes v1.16.2
-	k8s.io/utils v0.0.0-20191218082557-f07c713de883
+	k8s.io/utils v0.0.0-20200324210504-a9aa75ae1b89
 )
 
 replace (
-	k8s.io/api => k8s.io/api v0.17.5
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.17.5
-	k8s.io/apimachinery => k8s.io/apimachinery v0.17.6-beta.0
-	k8s.io/apiserver => k8s.io/apiserver v0.17.5
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.17.5
-	k8s.io/client-go => k8s.io/client-go v0.17.5
-	k8s.io/cloud-provider => k8s.io/cloud-provider v0.17.5
-	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.17.5
-	k8s.io/code-generator => k8s.io/code-generator v0.17.6-beta.0
-	k8s.io/component-base => k8s.io/component-base v0.17.5
-	k8s.io/cri-api => k8s.io/cri-api v0.17.6-beta.0
-	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.17.5
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.17.5
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.17.5
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.17.5
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.17.5
-	k8s.io/kubectl => k8s.io/kubectl v0.17.5
-	k8s.io/kubelet => k8s.io/kubelet v0.17.5
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.17.5
-	k8s.io/metrics => k8s.io/metrics v0.17.5
+	k8s.io/api => k8s.io/api v0.18.3
```

Comment:
Is this `replace` necessary, given that `k8s.io/api v0.18.3` is in the `require` above?

Reply:
It should not be, and a few other direct dependencies of ours probably don't need the replacement either. I think it's a consequence of how we upgrade dependencies through [our script](https://github.com/digitalocean/csi-digitalocean/blob/275ffede814edfa5ffa722a8ef1adc6a61d02217/scripts/update-k8s.sh), which we need because we have to vendor k8s.io/kubernetes for a resizing library, and that requires replacing all of the monorepo's dependencies in the fashion we do.

We could exclude the direct dependencies from the script, but it'd involve a bit effort for I think relatively little gain. AFAIK, upstream is planning on moving the resize logic into a separate repository as well; at that point, we can simplify our dependency management process by referencing direct dependencies only and letting gomod pull in just the transitive dependencies we need.